aesara/gpuarray/elemwise.py

```diff
@@ -2955,6 +2955,7 @@ def c_headers(self, **kwargs):
     def c_code(self, node, name, inp, out, sub):
         (x,) = inp
         (z,) = out
+        # TODO FIXME: Make this work without strict `Type` equality
         if node.inputs[0].type in complex_types:
```

Comment:
I presume over here (and at all the other similar TODOs) we'd need to check something like `node.inputs[0].type.is_compatible(some_type)` for `some_type` in `complex_types` 

I may be misunderstanding the issue over here though.

Reply:
Those specific TODOs need to be addressed, I haven't gotten to them yet.  At this point, I only want to identify all the places where strict `Type` comparisons are made.
  
More importantly, a lot of the changes in that one commit (i.e. a670243) need to carefully reconsidered, because `Type.is_compatible` is not symmetric (i.e. `t1.is_compatible(t2)` is not equal to `t2.is_compatible(t1)` unless `t1 == t2`), and some of those order choices might not be correct.